src/v/container/contiguous_range_map.h

```diff
@@ -293,11 +293,9 @@ class contiguous_range_map {
     }
 
 private:
-    static void check_key(KeyT key) {
-        vassert(
-          key >= 0,
-          "contiguous_range_map keys must be positive. current key {}",
-          key);
+    static bool valid_key(KeyT key) {
+        // map keys must be positive.
+        return key >= 0;
```

Comment:
why use a generic type if it must be convertible to int?

Reply:
Template definition restricts it to integral types

```
template<std::integral KeyT, typename ValueT>
class contiguous_range_map {
```

Reply:
yeh i guess if we had a fixed integral type then we'd end up with various narrowing conversion warnings etc...